### PR TITLE
HSEARCH-3086 Search 6 groundwork - Add the missing non-field projections compared to Search 5

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3086

### CODE REVIEW

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/ElasticsearchExtension.java [hunk 1]
```diff
@@ -35,18 +41,27 @@
  * <strong>WARNING:</strong> while this type is API, because instances should be manipulated by users,
  * all of its methods are considered SPIs and therefore should never be called directly by users.
  * In short, users are only expected to get instances of this type from an API and pass it to another API.
+ *
+ * @param <R> The reference type for projections.
+ * Users should not have to care about this, as the parameter will automatically take the appropriate value when calling
+ * {@code .extension( Lucene.get() }.
+ * @param <O> The loaded object type for projections.
+ * Users should not have to care about this, as the parameter will automatically take the appropriate value when calling
+ * {@code .extension( Lucene.get() }.
```
Comment (gsmet) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/ElasticsearchExtension.java]: Same here.

File: backend/lucene/src/main/java/org/hibernate/search/backend/lucene/LuceneExtension.java [hunk 2]
```diff
@@ -35,18 +41,27 @@
  * <strong>WARNING:</strong> while this type is API, because instances should be manipulated by users,
  * all of its methods are considered SPIs and therefore should never be called directly by users.
  * In short, users are only expected to get instances of this type from an API and pass it to another API.
+ *
+ * @param <R> The reference type for projections.
+ * Users should not have to care about this, as the parameter will automatically take the appropriate value when calling
+ * {@code .extension( Lucene.get() }.
+ * @param <O> The loaded object type for projections.
+ * Users should not have to care about this, as the parameter will automatically take the appropriate value when calling
+ * {@code .extension( Lucene.get() }.
```
Comment (gsmet) [inline backend/lucene/src/main/java/org/hibernate/search/backend/lucene/LuceneExtension.java]: I think it's `LuceneExtension.get()` - 2 occurrences above.

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchBackendFactory.java [hunk 3]
```diff
@@ -100,14 +101,16 @@
 					DefaultGsonProvider.create( this::createES5GsonBuilderBase, logPrettyPrinting );
 			client.init( dialectSpecificGsonProvider );
 
+			Gson userFacingGson = new Gson();
```
Comment (gsmet) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchBackendFactory.java]: Maybe let's enable pretty-printing.

**Comment by yrodiere** [comment]
Addressed your comments and merged. Thanks!

**Pull request merged**